Tools/SandboxTest.ps1

```diff
@@ -273,6 +305,132 @@ function Test-FileChecksum {
+#   token is re-checked. If the file has content, the date is checked to see if the token is expired. This can't catch every
+#   edge case, but it should catch a majority of the use cases.
+####
+function Test-GithubToken {
+    param (
+        [Parameter(Mandatory = $true)]
+        [AllowEmptyString()]
+        [String] $Token
+    )
+
+    # If the token is empty, there is no way that it can be valid
+    if ([string]::IsNullOrWhiteSpace($Token)) { return $false }
+
+    Write-Verbose 'Hashing GitHub Token'
+    $_memoryStream = [System.IO.MemoryStream]::new()
+    $_streamWriter = [System.IO.StreamWriter]::new($_memoryStream)
+    $_streamWriter.Write($Token)
+    $_streamWriter.Flush()
+    $_memoryStream.Position = 0
+
+    $tokenHash = Get-FileHash -InputStream $_memoryStream | Select-Object -ExpandProperty Hash
+
+    # Dispose of the reader and writer for hashing the token to ensure they cannot be accessed outside of the intended scope
+    Write-Debug 'Disposing of hashing components'
+    $_streamWriter.DisposeAsync() 1> $null
+    $_memoryStream.DisposeAsync() 1> $null
+
+    # Check for the cached token file
+    Initialize-Folder -FolderPath $script:TokenValidationCache | Out-Null
+    $cachedToken = Get-ChildItem -Path $script:TokenValidationCache -Filter $tokenHash -ErrorAction SilentlyContinue
+
+    if ($cachedToken) {
+        Write-Verbose 'Token was found in the cache'
+        # Check the age of the cached file
+        $cachedTokenAge = (Get-Date) - $cachedToken.LastWriteTime | Select-Object -ExpandProperty TotalDays
+        $cachedTokenAge = [Math]::Round($cachedTokenAge, 2) # We don't need all the precision the system provides
+        Write-Debug "Token has been in the cache for $cachedTokenAge days"
+        $cacheIsExpired = $cachedTokenAge -ge $script:CachedTokenExpiration
+        $cachedTokenContent = (Get-Content $cachedToken -Raw).Trim() # Ensure any trailing whitespace is ignored
+        $cachedTokenIsEmpty = [string]::IsNullOrWhiteSpace($cachedTokenContent)
+
+        # It is possible for a token to be both empty and expired. Since these are debug and verbose messages, showing both doesn't hurt
+        if ($cachedTokenIsEmpty) { Write-Verbose 'Cached token had no content. It will be re-validated' }
+        if ($cacheIsExpired) { Write-Verbose "Cached token is older than $script:CachedTokenExpiration days. It will be re-validated" }
+
+        if (!$cacheIsExpired -and !$cachedTokenIsEmpty) {
+            # Check the content of the cached file in case the actual token expiration is known
+            Write-Verbose 'Attempting to fetch token expiration from cache'
+            # Since Github adds ` UTC` at the end, it needs to be stripped off. Trim is safe here since the last character should always be a digit
+            $cachedExpirationForParsing = $cachedTokenContent.TrimEnd(' UTC')
```

Comment:
Since we use `ToLongDateString()` when saving the token, it shouldn't have ` UTC` anymore at the end

Reply:
True; I'll lleave this just in case though